dot/core/service.go

```diff
@@ -551,3 +579,27 @@ func (s *Service) GetReadProofAt(block common.Hash, keys [][]byte) (
 
 	return block, proofForKeys, nil
 }
+
+// buildExternalTransaction builds an external transaction based on the current TransactionQueueAPIVersion
+// See https://github.com/paritytech/substrate/blob/polkadot-v0.9.25/primitives/transaction-pool/src/runtime_api.rs#L25-L55
+func (s *Service) buildExternalTransaction(rt runtime.Instance, ext types.Extrinsic) (types.Extrinsic, error) {
+	runtimeVersion := rt.Version()
+	txQueueVersion := runtimeVersion.TaggedTransactionQueueVersion(runtimeVersion)
+	var externalExt types.Extrinsic
+	switch txQueueVersion {
+	case supportedTxVersion:
+		externalExt = types.Extrinsic(concatenateByteSlices([][]byte{
+			{byte(types.TxnExternal)},
+			ext,
+			s.blockState.BestBlockHash().ToBytes(),
+		}))
```

Comment:
```suggestion
extrinsics := [][]byte{{byte(types.TxnExternal)}, ext, s.blockState.BestBlockHash().ToBytes()}
externalExt = types.Extrinsic(bytes.Join(extrinsics, nil))
```

Reply:
nit it's not really `extrinsics`, it's more of a `extrinsicParts` I guess